test/dotnet-add-p2p.Tests/GivenDotnetAddP2P.cs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using Microsoft.DotNet.Cli.Utils;
+using Xunit;
+using FluentAssertions;
+
+namespace Microsoft.DotNet.Cli.Add.P2P.Tests
+{
+    public class GivenDotnetAddP2P : TestBase
+    {
+        private static readonly string AppPath = Path.Combine("App", "App.csproj");
+        private static readonly string Lib1Path = Path.Combine("Lib1", "Lib1.csproj");
+        private static readonly string Lib2Path = Path.Combine("Lib2", "Lib2.csproj");
+        private static readonly string Lib3Path = Path.Combine("Lib3", "Lib3.csproj");
+        private static readonly string Lib4Path = Path.Combine("Lib4", "Lib4.csproj");
+
+        private string Setup([System.Runtime.CompilerServices.CallerMemberName] string callingMethod = nameof(Setup))
+        {
+            const string TestGroup = "NonRestoredTestProjects";
+            const string ProjectName = "DotnetAddP2PProjects";
+            return GetTestGroupTestAssetsManager(TestGroup).CreateTestInstance(ProjectName, callingMethod: callingMethod).Path;
+        }
+
+        [Theory]
+        [InlineData("--help")]
+        [InlineData("-h")]
+        public void WhenHelpOptionIsPassedItPrintsHelp(string helpArg)
+        {
+            var cmd = new AddP2PCommand().Execute(helpArg);
+            cmd.Should().Pass();
+            cmd.StdOut.Should().Contain("Usage");
+        }
+
+        [Theory]
+        [InlineData("idontexist")]
+        [InlineData("Lib1")] // folder name
+        [InlineData("Broken.csproj")] // exists but broken
+        public void WhenInvalidProjectIsPassedItFails(string projName)
+        {
+            string testRoot = Setup();
+            new AddP2PCommand()
+                    .WithWorkingDirectory(testRoot)
+                    .Execute($"{projName} -r {Lib1Path}")
+                .ExitCode.Should().NotBe(0);
```

Comment:
I was thinking of doing that for a while but since we are planning to add localization of strings soon, I don't believe this is a good idea as we will need to make sure the tests will run with en-US locale